zerver/lib/actions.py

```diff
@@ -275,16 +275,69 @@
     )
 
 
+class SubscriptionStreamDict(TypedDict, total=False):
```

Comment:
Ideally we should avoid `total=False`, which affects all keys, in favor of `NotRequired[]` on specific keys ([PEP 655](https://peps.python.org/pep-0655/), supported in [mypy 0.930](https://mypy-lang.blogspot.com/2021/12/mypy-0930-released.html)).

Or if this is only there to let us construct these dicts by mutation:

```python3
    result = SubscriptionStreamDict()
    result["date_created"] = datetime_to_timestamp(raw_stream_dict["date_created"])
    result["description"] = raw_stream_dict["description"]
    result["first_message_id"] = raw_stream_dict["first_message_id"]
    # …
```

we should instead construct them functionally:

```python3
    result = SubscriptionStreamDict(
        date_created=datetime_to_timestamp(raw_stream_dict["date_created"]),
        description=raw_stream_dict["description"],
        first_message_id=raw_stream_dict["first_message_id"],
        # …
    )
```

That way mypy will tell us if we forgot a required key.